noir-projects/noir-protocol-circuits/bootstrap.sh

```diff
@@ -79,48 +77,48 @@ function compile {
     cache_upload circuit-$hash.tar.gz $json_path &> /dev/null
   fi
 
-  if echo "$name" | grep -qE "${private_tail_regex}"; then
-    local proto="client_ivc_tail"
-    # We still need the standalone IVC vk. We also create the final IVC vk from the tail (specifically, the number of public inputs is used from it).
-    local write_vk_cmd="write_vk --scheme client_ivc --verifier_type standalone"
-  elif echo "$name" | grep -qE "${ivc_regex}"; then
-    local proto="client_ivc"
-    local write_vk_cmd="write_vk --scheme client_ivc --verifier_type standalone"
-  elif echo "$name" | grep -qE "${rollup_honk_regex}"; then
-    local proto="ultra_rollup_honk"
-    local write_vk_cmd="write_vk --scheme ultra_honk --ipa_accumulation"
-  elif echo "$name" | grep -qE "rollup_root"; then
-    local proto="ultra_keccak_honk"
-    # the root rollup does not need to inject a fake ipa claim
-    # and does not need to inject a default agg obj, so no -h flag
-    local write_vk_cmd="write_vk --scheme ultra_honk --oracle_hash keccak"
-  else
-    local proto="ultra_honk"
-    local write_vk_cmd="write_vk --scheme ultra_honk"
-  fi
   # No vks needed for simulated circuits.
   [[ "$name" == *"simulated"* ]] && return
 
-  # TODO: Change this to add verification_key to original json, like contracts does.
-  # Will require changing TS code downstream.
-  bytecode_hash=$(jq -r '.bytecode' $json_path | sha256sum | tr -d ' -')
-  hash=$(hash_str "$BB_HASH-$bytecode_hash-$proto-$(cache_content_hash "^noir-projects/noir-protocol-circuits/bootstrap.sh")")
+  # Add verification key to original json, similar to contracts.
+  # This adds keyAsBytes and keyAsFields to the JSON artifact.
+  local bytecode_hash=$(jq -r '.bytecode' $json_path | sha256sum | tr -d ' -')
+  local hash=$(hash_str "$BB_HASH-$bytecode_hash-$name-3")
+  local key_path="$key_dir/$name.vk.data.json"
   if ! cache_download vk-$hash.tar.gz 1>&2; then
-    local key_path="$key_dir/$name.vk.data.json"
-    echo_stderr "Generating vk for function: $name..."
     SECONDS=0
-    outdir=$(mktemp -d)
+    local outdir=$(mktemp -d)
     trap "rm -rf $outdir" EXIT
-    local vk_cmd="jq -r '.bytecode' $json_path | base64 -d | gunzip | $BB $write_vk_cmd -b - -o $outdir"
-    echo_stderr $vk_cmd
-    dump_fail "$vk_cmd"
+    function write_vk {
+      if echo "$name" | grep -qE "${private_tail_regex}"; then
```

Comment:
I updated this naming in some PR but maybe it hasn't been mrged yet - as long as the private_tail_regex is finding the hiding kernel then this should be good